### PR TITLE
Add 2.2.0 announcement post

### PR DESCRIPTION
**Pull request by timriley**
[Readable post](https://github.com/hanami/hanami.github.io/blob/220-announcement/source/blog/2024-11-05-hanami-220.html.markdown)

### CODE REVIEW

File: source/blog/2024-11-05-hanami-220.html.markdown [hunk 1]
```diff
@@ -0,0 +1,244 @@
+---
+title: "Hanami 2.2.0: Persistence pays off"
```
Comment (cllns) [inline source/blog/2024-11-05-hanami-220.html.markdown]:
```suggestion
title: "Hanami 2.2: Persistence pays off"
```
I think it's cleaner and looks better, plus it matches the format for 2.1 and 2.0 releases.

I really like the name "Persistence pays off". Is there some way we could add a reference to "Operation" too? I was thinking "Operation Persistence is complete" but [that's already the name of something tragic](https://en.wikipedia.org/wiki/Operation_Persistence).

File: source/blog/2024-11-05-hanami-220.html.markdown [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+  change do
+    create_table :posts do
+      primary_key :id
+      column :title, :text, null: false
+    end
+  end
+end
+```
+
+Now you can migrate your database with `hanami db migrate`.
```
Comment (cllns) [inline source/blog/2024-11-05-hanami-220.html.markdown:36]:
```suggestion
Now you can migrate your database with `bundle exec hanami db migrate`.
```

File: source/blog/2024-11-05-hanami-220.html.markdown [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+
+After this, you can generate a new relation: `hanami generate relation posts`. Relations describe your low-level data sources. Here, this means your database tables. Relations are also your place to add reusable, chainable methods that you can use as the building blocks for expressive, higher-level queries. Add something simple to get started:
```
Comment (cllns) [inline source/blog/2024-11-05-hanami-220.html.markdown]:
```suggestion
After this, you can generate a new relation: `bundle exec hanami generate relation posts`. Relations describe your low-level data sources. Here, this means your database tables. Relations are also your place to add reusable, chainable methods that you can use as the building blocks for expressive, higher-level queries. Add something simple to get started:
```

File: source/blog/2024-11-05-hanami-220.html.markdown [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+
+      def order_by_latest
+        order(self[:id].desc)
+      end
+    end
+  end
+end
+```
+
+Relations are the primary component of your app’s database layer. While you can interact with them directly, it’s better to build a repo. With repos, you get to build your very own database interface, so you can better manage how your data is accessed across your app. You can build a post repo with `hanami generate repo posts`. Here you can define a method to return your latest posts, built using the methods in your relation:
```
Comment (cllns) [inline source/blog/2024-11-05-hanami-220.html.markdown]: Is it true that relations are "primary"? That implies other components are "secondary", etc. I guess so, in the sense that one could get by using only Relations.

File: source/blog/2024-11-05-hanami-220.html.markdown [hunk 5]
```diff
@@ -0,0 +1,244 @@
...
+        end
+      end
+    end
+  end
+end
+```
+
+Repo methods return structs: plain old value objects, just data, with no live connection back to the database. This means you can be confident in passing them all around your app, knowing things like accidental n+1 queries are a thing of the past.
+
+You can customize these structs by creating your own matching classes. Make one for your posts with `hanami generate struct post`. Inside these classes, you can access any of the attributes selected in your repo’s corresponding database query.
```
Comment (cllns) [inline source/blog/2024-11-05-hanami-220.html.markdown]:
```suggestion
You can optionally customize these structs by creating your own matching classes. Make one for your posts with `hanami generate struct post`. Inside these classes, you can access any of the attributes selected in your repo’s corresponding database query.
```
Not sure if this is necessary but might be clarifying

Comment (cllns) [inline source/blog/2024-11-05-hanami-220.html.markdown]:
```suggestion
You can customize these structs by creating your own matching classes. Make one for your posts with `bundle exec hanami generate struct post`. Inside these classes, you can access any of the attributes selected in your repo’s corresponding database query.
```

File: source/blog/2024-11-05-hanami-220.html.markdown [hunk 6]
```diff
@@ -0,0 +1,244 @@
...
+
+- Share a single database, but have each slice provide its own relations, so they can choose exactly how much of the database to expose. This is our default.
+- Use a dedicated database for certain slices. This is as easy as a `SLICE_NAME__DATABASE_URL` env var!
+- Connect to any number of _additional_ databases within your app or slice. This is as easy as creating a `SLICE_NAME__DATABASE_URL__GATEWAY_NAME` env var!
+- Adopt a simpler, blended development experience by sharing a single database and set of relations across all slices.
+- Or any combination of the above!
+
+## Smooth operations
+
+Along with Hanami 2.2, we’re excited to debut the brand new [dry-operation](https://dry-rb.org/gems/dry-operation) gem. With dry-operation, you have a streamlined tool for organizing your business logic into flexible, composable objects made from flows of internal steps.
```
Comment (cllns) [inline source/blog/2024-11-05-hanami-220.html.markdown]: Should we add something here about how dry-operation is a first-class part of the Hanami stack, but an optional one? People are welcome to use other libraries/their own

File: source/blog/2024-11-05-hanami-220.html.markdown [hunk 7]
```diff
@@ -0,0 +1,244 @@
...
+    end
+  end
+end
+```
+
+Operations are natively integrated with Hanami’s database layer, providing `transaction do ... end` blocks to ensure database changes are written together, with any intervening `Failure` automatically rolling back the transaction.
+
+dry-operation is the long-awaited successor to the venerable [dry-transaction](http://dry-rb.org/gems/dry-transaction) gem, and I’m deeply grateful to [Marc Busqué](https://github.com/waiting-for-dev) for building it.
+
+## More up our sleeve
```
Comment (cllns) [inline source/blog/2024-11-05-hanami-220.html.markdown:201]: Should we add something about supporting Ruby 3.1+?

Should we add a link to our Discourse, and maybe a reference to their being a chat on Discourse that poeple can see only once they create an account?

File: source/blog/2024-11-05-hanami-220.html.markdown (hunk 4 above)
Comment (cllns) [inline source/blog/2024-11-05-hanami-220.html.markdown]:
```suggestion
Relations are the primary component of your app’s database layer. While you can interact with them directly, it’s better to build a repo. With repos, you get to build your very own database interface, so you can better manage how your data is accessed across your app. You can build a post repo with `bundle exec hanami generate repo posts`. Here you can define a method to return your latest posts, built using the methods in your relation:
```

**Review by cllns — COMMENTED** [review]
Amazing! So exciting. Left a bunch of small comments/suggestions. 

I tried to change all relevant `hanami` commands to have `bundle exec` in front of them. Not strictly necessary since most Ruby users will know to do that or make binstubs etc but could be usful

File: source/blog/2024-11-05-hanami-220.html.markdown (hunk 4 above)
Comment (timriley) [reply source/blog/2024-11-05-hanami-220.html.markdown]: Good callout, this is confusing language. I've just gone and removed this leading sentence entirely! Let me know what you think.

File: source/blog/2024-11-05-hanami-220.html.markdown (hunk 5 above)
Comment (timriley) [reply source/blog/2024-11-05-hanami-220.html.markdown]: Yeah, I think it's good, added 👍🏼

File: source/blog/2024-11-05-hanami-220.html.markdown (hunk 7 above)
Comment (timriley) [reply source/blog/2024-11-05-hanami-220.html.markdown:201]: Good point, added "Dropped support for Ruby 3.0"

File: source/blog/2024-11-05-hanami-220.html.markdown (hunk 1 above)
Comment (timriley) [reply source/blog/2024-11-05-hanami-220.html.markdown]: Ah yes, thank you! Have updated the title.

In terms of the tag line, I'm glad you like "persistence pays off" :) This was a name I gave to solnic for a talk he did at RubyConf AU back in 2017, but I like it a lot and I figured this was the perfect time to bring it back.

Persistence is clearly the hero of this release, so I think it's fine for that to be the main focus of the tag line. We do make it clear in other places that operations are the next biggest thing :)

File: source/blog/2024-11-05-hanami-220.html.markdown (hunk 2 above)
Comment (timriley) [reply source/blog/2024-11-05-hanami-220.html.markdown:36]: Thanks for all these `bundle exec` suggestions. I'm in mixed minds about this. Prefixing the `bundle exec` makes everything feel a little more ungainly, and I do think Rubyists will understand whether they need it or not.

I think we have a compromise approach already in place: `bundle exec` _is_ already in place for the very first mention of the `hanami` command, just as an indicator that it's needed. The rest can live without it, for succinctness and ease of reading.

File: source/blog/2024-11-05-hanami-220.html.markdown (hunk 6 above)
Comment (timriley) [reply source/blog/2024-11-05-hanami-220.html.markdown]: Added a paragraph on this. Let me know what you think.

**Pull request merged**